pkg/local_object_storage/blobstor/exists.go

```diff
@@ -29,20 +33,35 @@ func (r ExistsRes) Exists() bool {
 func (b *BlobStor) Exists(prm *ExistsPrm) (*ExistsRes, error) {
 	// check presence in shallow dir first (cheaper)
 	exists, err := b.existsBig(prm.addr)
+
+	// If there was an error during existence check below,
+	// it will be returned unless object was found in blobovnicza.
+	// Otherwise, it is logged and the latest error is returned.
+	// FSTree    | Blobovnicza | Behaviour
+	// found     | (not tried) | return true, nil
+	// not found | any result  | return the result
+	// error     | found       | log the error, return true, nil
+	// error     | not found   | return the error
+	// error     | error       | log the first error, return the second
 	if !exists {
-		// TODO: #1143 do smth if err != nil
+		var smallErr error
 
-		// check presence in blobovnicza
-		exists, err = b.existsSmall(prm.addr)
+		exists, smallErr = b.existsSmall(prm.addr)
+		if err != nil && (smallErr != nil || exists) {
+			b.log.Warn("error occured during object existence checking",
+				zap.Stringer("address", prm.addr),
+				zap.String("error", err.Error()))
```

Comment:
`zap.Error(err)`?

Reply:
We mostly use `zap.String` in engine. `zap.Error` is more verbose.